src/coreclr/CMakeLists.txt

```diff
@@ -109,7 +109,7 @@ if(CLR_CMAKE_HOST_UNIX)
         endif()
     endif()
 
-    if(CLR_CMAKE_TARGET_ALPINE_LINUX AND CLR_CMAKE_TARGET_ARCH_I386)
+    if(CLR_CMAKE_TARGET_ALPINE_LINUX)
```

Comment:
For the riscv64 code, only the following did the trick:
```diff
diff --git a/src/runtime/eng/native/configurecompiler.cmake b/src/runtime/eng/native/configurecompiler.cmake
index f5500b885..67794f1db 100644
--- a/src/runtime/eng/native/configurecompiler.cmake
+++ b/src/runtime/eng/native/configurecompiler.cmake
@@ -191,7 +191,7 @@ endif(CLR_CMAKE_HOST_UNIX)
 if(CLR_CMAKE_HOST_LINUX)
   add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
   add_linker_flag(-Wl,--build-id=sha1)
-  add_linker_flag(-Wl,-z,relro,-z,now)
+  add_linker_flag(-Wl,-z,relro,-z,now,-z,notext)
 elseif(CLR_CMAKE_HOST_FREEBSD)
   add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
   add_linker_flag("-Wl,--build-id=sha1")
```
That said, this should be, at best, a workaround. Packaging guidelines on Alpine Linux strongly discourage textrels. It would be nice to find the source of this for `linux-musl-x86`. For riscv64, I've notified the PR owner.

Reply:
riscv64 port isn't completed, so I'm not worried about it as part of this PR.

> It would be nice to find the source of this for `linux-musl-x86`.

I suspect it is due to libucontext doing something different than OOTB ucontext support on other architectures. This PR is only silencing the warning.

Reply:
The piece of code in libunwind that uses libucontext is https://github.com/dotnet/runtime/blob/edb161ab06ddc69d27aee2c0e990a60221ebbe92/src/native/external/libunwind/src/x86/Gos-linux.c#L309
Removing this line allowed me to build runtime for x86 without linking to libucontext. The textrel issues persists.